From 9b3a68992fdaa85c14399a88f5a1d159def95549 Mon Sep 17 00:00:00 2001
From: Pablo Neira Ayuso <pablo@netfilter.org>
Date: Wed, 18 Oct 2023 22:20:10 +0200
Subject: [PATCH 03/18] netfilter: nft_set_pipapo: no need to call
 pipapo_deactivate() from flush

mainline inclusion
from mainline-v6.7-rc1
commit 26cec9d4144eb23c45cd5c033d5c141f04d61a9c
category: bugfix
bugzilla: https://atomgit.com/src-openeuler/kernel/issues/17278
CVE: CVE-2026-72252

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=26cec9d4144eb23c45cd5c033d5c141f04d61a9c

--------------------------------

Use the element object that is already offered instead.

Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Dong Chenchen <superdcc97@163.com>
---
 net/netfilter/nft_set_pipapo.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/net/netfilter/nft_set_pipapo.c b/net/netfilter/nft_set_pipapo.c
index 408aaa579c78..73396cb73549 100644
--- a/net/netfilter/nft_set_pipapo.c
+++ b/net/netfilter/nft_set_pipapo.c
@@ -1873,8 +1873,9 @@ static bool nft_pipapo_flush(const struct net *net, const struct nft_set *set,
 {
 	struct nft_pipapo_elem *e = elem;
 
-	return pipapo_deactivate(net, set, (const u8 *)nft_set_ext_key(&e->ext),
-				 &e->ext);
+	nft_set_elem_change_active(net, set, &e->ext);
+
+	return true;
 }
 
 /**
-- 
2.43.0

